ako-infra/ingestion/vcf_k8s_controller.go

```diff
@@ -134,87 +134,88 @@ func (c *VCFK8sController) AddNCPSecretEventHandler(k8sinfo K8sinformers, stopCh
+			c.handleNamespaceAdd()
 		},
 		UpdateFunc: func(old, obj interface{}) {
-			utils.AviLog.Infof("NSX Network Config Update Event")
+			utils.AviLog.Infof("Namespace Update Event")
 		},
 		DeleteFunc: func(obj interface{}) {
-			utils.AviLog.Infof("NSX Network Config Delete Event")
+			utils.AviLog.Infof("Namespace Delete Event")
 			crd := obj.(*unstructured.Unstructured)
 			_, found, err := unstructured.NestedStringMap(crd.UnstructuredContent(), "spec")
 			if err != nil || !found {
-				utils.AviLog.Warnf("nsxnetworkconfiguration spec not found: %+v", err)
+				utils.AviLog.Warnf("Namespace spec not found: %+v", err)
 				return
 			}
-			handleNSXNetworkConfigDelete()
+			c.handleNamespaceDelete()
 		},
 	}
-	c.dynamicInformers.NSXNetworkConfigInformer.Informer().AddEventHandler(nsxNetworkConfigHandler)
+	c.informers.NSInformer.Informer().AddEventHandler(namespaceHandler)
 
-	go c.dynamicInformers.NSXNetworkConfigInformer.Informer().Run(stopCh)
-	if !cache.WaitForCacheSync(stopCh, c.dynamicInformers.NSXNetworkConfigInformer.Informer().HasSynced) {
+	go c.informers.NSInformer.Informer().Run(stopCh)
+	if !cache.WaitForCacheSync(stopCh, c.informers.NSInformer.Informer().HasSynced) {
 		runtime.HandleError(fmt.Errorf("Timed out waiting for caches to sync"))
 	} else {
-		utils.AviLog.Info("Caches synced for nsxnetworkconfiguration informer")
+		utils.AviLog.Info("Caches synced for Namespace informer")
 	}
 }
 
-func addNSXNetworkConfigCount() error {
-	dynamicClient := lib.GetVCFDynamicClientSet()
-	crdClient := dynamicClient.Resource(lib.NSXNetworkConfigGVR)
-	crdList, err := crdClient.List(context.TODO(), metav1.ListOptions{})
+func (c *VCFK8sController) addWorkloadNamespaceCount() error {
+	clientSet := c.informers.ClientSet
+	nsList, err := clientSet.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
 	if err != nil {
 		return err
 	}
 
-	NsxNetworkConfigCount = len(crdList.Items)
+	workloadNSCount := getWorkloadNamespaceCount(nsList)
+	WorkloadNamespaceCount = workloadNSCount
 	return nil
 }
 
-func handleNSXNetworkConfigAdd() {
-	dynamicClient := lib.GetVCFDynamicClientSet()
-	crdClient := dynamicClient.Resource(lib.NSXNetworkConfigGVR)
-	crdList, err := crdClient.List(context.TODO(), metav1.ListOptions{})
+func (c *VCFK8sController) handleNamespaceAdd() {
+	clientSet := c.informers.ClientSet
+	nsList, err := clientSet.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
 	if err != nil {
 		utils.AviLog.Error(err)
 		return
 	}
 
+	workloadNSCount := getWorkloadNamespaceCount(nsList)
+
 	// Only when before the addition, the count was 0, (and now it becomes more than 0),
 	// we must reconfigure the SEG, by rebooting AKO. On reboot AKO ensures SEG configuration.
-	if NsxNetworkConfigCount == 0 && len(crdList.Items) > 0 {
-		utils.AviLog.Fatalf("First NSXNetworkConfigurations added in cluster. Rebooting AKO for infra configuration.")
+	if WorkloadNamespaceCount == 0 && workloadNSCount > 0 {
```

Comment:
I think the variable names WorkloadNamespaceCount and workloadNSCount are too mimilar and may get confusing later. Lets check if we can change the names.